src/web/src/app/space/list/component.ts

```diff
@@ -0,0 +1,262 @@
+
+	@ViewChild('invitationsPaginator') invitationsPaginator: MatPaginator;
+	@ViewChild('linksPaginator') linksPaginator: MatPaginator;
+	@ViewChild('spacesPaginator') spacesPaginator: MatPaginator;
+	@ViewChild('scrollToInvitationForm') scrollToInvitationForm: ElementRef;
+	@ViewChild('scrollToLinkForm') scrollToLinkForm: ElementRef;
+
+	constructor(
+		@Inject(DOCUMENT) private document: any,
+		private fb: FormBuilder,
+		private space: SpaceService,
+		private user: UserService,
+	) {
+		this._initInvitationForm();
+		this._initLinkForm();
+	}
+
+	ngOnInit() {
+		this.currentUserId = this.user.info.id;
+
+		const pageSizeStr = localStorage.getItem('spaceListPageSize');
+		const pageSize = JSON.parse(pageSizeStr);
+		if (pageSize?.invitations && pageSize?.links && pageSize?.spaces){
+			this.pageSize = pageSize;
+		}
+
+		this.getInvitations(this.pageSize.invitations, 0);
+		this.getLinks(this.pageSize.links, 0);
+		this.getSpaces(this.pageSize.spaces, 0);
+	}
+
+	getSpaces(limit: number, page: number) {
+		this.savePageSize("spaces", limit);
+		this.space.getList(limit, page)
+			.subscribe(res => {
+				this.spaces = res.results;
+				this.spacesTotal = res.count;
+			});
+	}
+
+	getInvitations(limit: number, page: number) {
+		this.savePageSize("invitations", limit);
+		this.space.getInvitationList(limit, page)
+			.subscribe(res => {
+				this.invitations = res.results as SpaceInvitationDetail[];
+				this.invitationsTotal = res.count;
+				this.space.fillInvitationDetailFields(this.invitations);
+			});
+	}
+
+	getLinks(limit: number, page: number) {
+		this.savePageSize("links", limit);
+		this.space.getLinkList(limit, page)
+			.subscribe(res => {
+				this.links = res.results as SpaceLinkDetail[];
+				this.linksTotal = res.count;
+				this.space.fillLinkDetailFields(this.links);
+			});
+	}
+
+	onLinkCopyBtn(link: SpaceLink){
+		let copyToClipboard = this.document.createElement('textarea');
+		copyToClipboard.style.position = 'fixed';
+		copyToClipboard.style.left = '0';
+		copyToClipboard.style.top = '0';
+		copyToClipboard.style.opacity = '0';
+		copyToClipboard.value = this.document.location.origin + this.linksURL + link.id;
+		document.body.appendChild(copyToClipboard);
+		copyToClipboard.focus();
+		copyToClipboard.select();
+		document.execCommand('copy');
+		document.body.removeChild(copyToClipboard);
+	}
+
+	onInvitationDelBtn(invitation: SpaceInvitation){
+		this.space.delInvitationById(invitation.id)
+			.subscribe(res => {
+				if (this.invitationsPaginator.pageIndex == 0){
+					this.getInvitations(this.pageSize.invitations, 0);
+				} else {
+					this.invitationsPaginator.firstPage();
+				}
+			}, err => console.warn(err));
+	}
+
+	onLinkDelBtn(link: SpaceLink){
+		this.space.delLinkById(link.id)
+			.subscribe(res => {
+				if (this.linksPaginator.pageIndex == 0){
+					this.getLinks(this.pageSize.links, 0);
+				} else {
+					this.linksPaginator.firstPage();
+				}			
+			}, err => console.warn(err));
```

Comment:
Ошибка уже логируется в requestWatcher. Тут это больше не нужно.
```suggestion
			});
```

Reply:
Убрал.